Add GET /hosted-payments/{id} endpoint

punit-shah-cko opened the pull request:
### Description of changes in PR

[PI-1355]:
- Add `id` & `_links.self` properties to `POST /hosted-payments` response
- Add `_links.self` property to `POST /payment-links` response

[PI-1358]:
- Add spec for new `GET /hosted-payments/{id}` endpoint

Also:
- Minor fixes for Payment Links spec

## Technical Writer

[//]: # 'Tag your Technical Writer'
@chrisi-webster-cko 

[PI-1355]: https://checkout.atlassian.net/browse/PI-1355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PI-1358]: https://checkout.atlassian.net/browse/PI-1358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ben-ahmady-cko commented on nas_spec/swagger.yaml:
```diff
@@ -46,12 +46,13 @@ info:
-     | 2021/10/12    |  Added Hosted Payments Page and Payment Links to specification.   |
+      | Date       | Description of change                                                         |
+      | ---------- | ----------------------------------------------------------------------------- |
+      | 2021/11/?? | Added specification for "Get Hosted Payments Page details" endpoint.          |
```
Please change the `??` when this is ready to go live 😅

ben-ahmady-cko commented on abc_spec/swagger.yaml:
```diff
@@ -49,11 +49,12 @@ info:
-     | 2021/10/18    |  Added `recommendation_code` to payment response.                 |
+      | Date       | Description of change                                                |
+      | ---------- | -------------------------------------------------------------------- |
+      | 2021/11/?? | Added specification for "Get Hosted Payments Page details" endpoint. |
```
Please change the `??` when this is ready to go live 😅

ben-ahmady-cko reviewed (approved):
Hey @punit-shah-cko the build looks good but I'm not familiar with the subject matter so relying on the approval you've had from Tom for that—and please make sure to change the changelog entries before merging 🙏

chrisi-webster-cko commented:
@punit-shah-cko Does your minor fix relate to PI-1370?

punit-shah-cko commented:
> @punit-shah-cko Does your minor fix relate to PI-1370?

@chrisi-webster-cko Not exactly - they're just minor changes to the existing spec (see the changes in commit d5e26dd):
- Made descriptions in the yaml file more readable (this doesn't change the output)
- Changed the `payment_id` field in the Get Payment Link details Active/Expired response examples to be `null` instead of the string `"undefined"`
  before: ![Screenshot 2021-11-15 at 14 49 25](https://user-images.githubusercontent.com/58436634/141803549-3e398ebc-e260-488d-9a38-ab6dd3d53e1a.png), after: ![Screenshot 2021-11-15 at 14 51 20](https://user-images.githubusercontent.com/58436634/141803606-d7ffe717-b073-4968-9fad-f895aee20cd6.png)
- Removed unnecessary parentheses from the regex pattern for the `PaymentLinkId` schema
- Show the `payment` and `payment_actions` links for the Get Payment Link details Payment Received response example only (and not the Active/Expired examples)
- Use the `PaymentLinkId` schema for the `id` property in the Create a Payment Link response
- Removed the `Cko-Request-Id` and `Cko-Version` headers from the Get Payment Link details response, as we don't actually include these headers in our response

ben-ahmady-cko reviewed (approved):
If Chrisi/Tom are happy, all good (Chrisi can't approve any more as a contributor)
